include/nbl/builtin/shader/loader/gltf/vertex_impl.glsl

```diff
@@ -0,0 +1,85 @@
+#include "nbl/builtin/shader/loader/gltf/common.glsl"
+
+#ifndef _NBL_VERT_INPUTS_DEFINED_
+#define _NBL_VERT_INPUTS_DEFINED_
+
+layout (location = _NBL_V_IN_POSITION_ATTRIBUTE_ID) in vec3 vPos;
+layout(location = _NBL_V_IN_NORMAL_ATTRIBUTE_ID) in vec3 vNormal;
+
+#ifndef _DISABLE_COLOR_ATTRIBUTES
+layout(location = _NBL_V_IN_COLOR_ATTRIBUTE_BEGINING_ID) in vec4 vColor_0;
+#ifdef _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_1_ENABLE
+#define _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_1 _NBL_V_IN_COLOR_ATTRIBUTE_BEGINING_ID + 1
+layout(location = _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_1) in vec4 vColor_1
+#endif // _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_1_ENABLE
+#ifdef _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_2_ENABLE
+#define _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_2 _NBL_V_IN_COLOR_ATTRIBUTE_BEGINING_ID + 2
+layout(location = _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_2) in vec4 vColor_2
+#endif // _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_2_ENABLE
+#endif // _DISABLE_COLOR_ATTRIBUTES
+
+#ifndef _DISABLE_UV_ATTRIBUTES
+layout (location = _NBL_V_IN_UV_ATTRIBUTE_BEGINING_ID) in vec2 vUV_0;
+#ifdef _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_1_ENABLE
+#define _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_1 _NBL_V_IN_UV_ATTRIBUTE_BEGINING_ID + 1
+layout (location = _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_1) in vec2 vUV_1
+#endif // _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_1_ENABLE
+#ifdef _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_2_ENABLE
+#define _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_2 _NBL_V_IN_UV_ATTRIBUTE_BEGINING_ID + 2
+layout(location = _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_2) in vec2 vUV_2
+#endif // _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_2_ENABLE
+#endif // _DISABLE_UV_ATTRIBUTES
```

Comment:
you can make arrays of attributes, just by the way. but three questions:
1. Does glTF stipulate using multiple vertex colours anywhere?
2. Does glTF stipulate using an RGBA or RGB vertex colour?
3. How many UV attributes do you actually need?